File: app/schemas/retrospective.py
from typing import Optional
from pydantic import BaseModel, ConfigDict
from datetime import datetime

# Properties to receive on item creation
class RetrospectiveCreate(BaseModel):
    title: str
    content: str
    evaluation_period_id: Optional[int] = None

# Properties to receive on item update
class RetrospectiveUpdate(BaseModel):
    title: Optional[str] = None
    content: Optional[str] = None

# Properties shared by models stored in DB
class RetrospectiveInDBBase(BaseModel):
    id: int
    user_id: int
    title: str
    content: Optional[str] = None
    evaluation_period_id: Optional[int] = None
    created_at: datetime
    updated_at: Optional[datetime] = None
    
    model_config = ConfigDict(from_attributes=True)

# Properties to return to client
class Retrospective(RetrospectiveInDBBase):
    pass

File: frontend/src/schemas/retrospective.ts
// frontend/src/schemas/retrospective.ts

export interface Retrospective {
  id: number;
  user_id: number;
  title: string;
  content: string;
  evaluation_period_id?: number | null;
  created_at: string;
  updated_at?: string | null;
}

export interface RetrospectiveCreate {
  title: string;
  content: string;
  evaluation_period_id?: number;
}

export interface RetrospectiveUpdate {
  title?: string;
  content?: string;
}

export interface GeneratedRetrospective {
  content: string;
}

export interface GenerateRetrospectivePayload {
  start_date: string;
  end_date: string;
}